sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/component_job.py

```diff
@@ -421,18 +417,12 @@ class DataTransferImportSchema(BaseNodeSchema):
     component = TypeSensitiveUnionField(
         {
             NodeType.DATA_TRANSFER: [
-                # inline component or component file reference starting with FILE prefix
-                NestedField(AnonymousDataTransferImportComponentSchema, unknown=INCLUDE),
-                # component file reference
-                DataTransferImportComponentFileRefField(),
+                # for registry type assets
+                RegistryStr(),
```

Comment:
this is wired. RegistryStr & ArmVersionedStr should be passed as `plain_union_fields`

Reply:
offline sink, will change to UnionField